pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -744,6 +742,9 @@ protected void buildSegmentForCommit(long buildTimeLeaseMs) {
       }
       _leaseExtender.addSegment(_segmentNameStr, buildTimeLeaseMs, _currentOffset);
       _segmentBuildDescriptor = buildSegmentInternal(true);
+    } catch (Exception e) {
```

Comment:
(major) I don't think we should catch the exception here. The exception should be thrown out

Reply:
yeah we can do a follow up refactor. this PR is specifically aiming at not changing the behavior of existing call flow.

Reply:
This is actually changing the behavior because the exception is not thrown out in the new code

Reply:
hmm. this is good catch!